test/e2e/tests/add-multiple-tokens.spec.js

```diff
@@ -0,0 +1,110 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  defaultGanacheOptions,
+  openDapp,
+  switchToNotificationWindow,
+  WINDOW_TITLES,
+  DAPP_URL,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Multiple ERC20 Watch Asset', function () {
+  // TODO: This assertion will change once the method is fixed.
+  it('should show multiple erc20 watchAsset token list, only confirms one bug', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await openDapp(driver, undefined, DAPP_URL);
+
+        // Create Token 1
+        await driver.clickElement({ text: 'Create Token', tag: 'button' });
+        await switchToNotificationWindow(driver);
+        await driver.findClickableElement({ text: 'Confirm', tag: 'button' });
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+
+        // Wait for token 1 address to populate in dapp
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await driver.wait(async () => {
+          const tokenAddressesElement = await driver.findElement(
+            '#tokenAddresses',
+          );
+          const tokenAddresses = await tokenAddressesElement.getText();
+          return tokenAddresses !== '';
+        }, 20000);
```

Comment:
20s seems long if we're only waiting 10s in the condition below this one

Reply:
Resolved in [604f46d](https://github.com/MetaMask/metamask-extension/pull/21381/commits/604f46dcd9cd0bf71892ba470a940515ac3191bf)